lib/registration.pm

```diff
@@ -459,6 +459,11 @@ sub get_addon_fullname {
 
 
 sub fill_in_reg_server {
+    # Set product specific SCC_REGCODE if it was provided. Defaults to SCC_REGCODE if set
+    my $scc_code = get_required_var('SCC_REGCODE') if get_var('SCC_REGCODE');
```

Comment:
No point to call get_required_var just after you check it's existance, you can simplify by calling just get_var instead.

Reply:
If I know perl correctly then it will only even *define* the variable if it is set, i.e. cause an ugly Perl warning otherwise.
I recommend to use just the second argument of get_var in both lines about the regcode setting, e.g.

```
my $regcode = get_var('SCC_REGCODE_' . uc(get_var('SLE_PRODUCT', '')), get_var ('SCC_REGCODE', ''));
...
type_string $regcode if $regcode;
```

Reply:
Should I also change this as well?

https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/registration.pm#L471

get_required_var() if get_var() was already in the code, so I left it that way even though it seemed redundant to me.

Reply:
sure, it's not about that you did something wrong . You can just improve code while modifying it